ethereum/api/src/test/java/org/hyperledger/besu/ethereum/api/jsonrpc/internal/privacy/methods/priv/PrivGetTransactionCountTest.java

```diff
@@ -56,4 +75,107 @@ public void verifyTransactionCount() {
+    final int returnedValue = Integer.decode((String) successResponse.getResult());
+    assertThat(returnedValue).isEqualTo(NONCE);
+  }
+
+  @Test
+  public void nonceProviderThrowsRuntimeExceptionProducesErrorResponse() {
+    final PrivateTransactionHandler privateNonceProvider = mock(PrivateTransactionHandler.class);
+    when(privateNonceProvider.getSenderNonce(any(), any())).thenThrow(RuntimeException.class);
+
+    final PrivacyGroup[] privacyGroups =
+        new PrivacyGroup[] {
+          new PrivacyGroup(
+              "Group1", PrivacyGroup.Type.PANTHEON, "Group1_Name", "Group1_Desc", new String[0]),
+          new PrivacyGroup(
+              "Group2", PrivacyGroup.Type.LEGACY, "Group2_Name", "Group2_Desc", new String[0]),
+        };
+    final Enclave enclave = mock(Enclave.class);
+    when(enclave.findPrivacyGroup(any())).thenReturn(privacyGroups);
+
+    final PrivGetTransactionCount method =
+        new PrivGetTransactionCount(parameters, privateNonceProvider, enclave);
+
+    final JsonRpcResponse response = method.response(legacyRequest);
+    assertThat(response).isInstanceOf(JsonRpcErrorResponse.class);
+
+    final JsonRpcErrorResponse errorResponse = (JsonRpcErrorResponse) response;
+    assertThat(errorResponse.getError())
+        .isEqualTo(JsonRpcError.GET_PRIVATE_TRANSACTION_NONCE_ERROR);
+  }
+
+  @Test
+  public void moreThanOneMatchingLegacyGroupProducesErrorResponse() {
+    final PrivateTransactionHandler privateNonceProvider = mock(PrivateTransactionHandler.class);
+    when(privateNonceProvider.getSenderNonce(any(), any())).thenThrow(RuntimeException.class);
+
+    final PrivacyGroup[] privacyGroups =
+        new PrivacyGroup[] {
+          new PrivacyGroup(
+              "Group1", PrivacyGroup.Type.LEGACY, "Group1_Name", "Group1_Desc", new String[0]),
+          new PrivacyGroup(
+              "Group2", PrivacyGroup.Type.LEGACY, "Group2_Name", "Group2_Desc", new String[0]),
+        };
+    final Enclave enclave = mock(Enclave.class);
+    when(enclave.findPrivacyGroup(any())).thenReturn(privacyGroups);
+
+    final PrivGetTransactionCount method =
+        new PrivGetTransactionCount(parameters, privateNonceProvider, enclave);
+
+    final JsonRpcResponse response = method.response(legacyRequest);
+    assertThat(response).isInstanceOf(JsonRpcErrorResponse.class);
+
+    final JsonRpcErrorResponse errorResponse = (JsonRpcErrorResponse) response;
+    assertThat(errorResponse.getError())
+        .isEqualTo(JsonRpcError.GET_PRIVATE_TRANSACTION_NONCE_ERROR);
+  }
+
+  @Test
+  public void noMatchingLegacyGroupReturnsNonce0() {
+    final PrivateTransactionHandler privateNonceProvider = mock(PrivateTransactionHandler.class);
+
+    final PrivacyGroup[] privacyGroups =
+        new PrivacyGroup[] {
+          new PrivacyGroup(
+              "Group1", PrivacyGroup.Type.PANTHEON, "Group1_Name", "Group1_Desc", new String[0]),
```

Comment:
Why is there `PrivacyGroup.Type.PANTHEON` and not `PrivacyGroup.Type.BESU`?

Reply:
Very good question! There is more stuff with "pantheon" in it's name. We should create a ticket for that. I wouldn't like to do that as part of this ticket.

Reply:
👌  I'll raise a PR to rename it now